server/src/main/java/org/apache/iotdb/db/query/expression/unary/TimeSeriesOperand.java

```diff
@@ -104,7 +109,7 @@ public IntermediateLayer constructIntermediateLayer(
 
       expressionIntermediateLayerMap.put(
           this,
-          memoryAssigner.getReference(this) == 1
+          memoryAssigner.getReference(this) == 1 || isPureConstantExpression()
```

Comment:
Why we construct a SingleInputColumnSingleReferenceIntermediateLayer when isPureConstantExpression() is true?

Reply:
That's a little bit tricky.

Since the constant point reader cannot provide a reasonable timestamp, using ElasticSerializableTVList with a constant point reader causes troubles. 

Also, I believe using a multi reference intermediate layer is mainly for avoiding duplicated IO and memory control, but unlike a time series reader, the constant reader won't cause any IO and the memory usage is little. 

Thus, I think SingleInputColumnSingleReferenceIntermediateLayer is a proper choice. What do you think?